californium-core/src/test/java/org/eclipse/californium/core/network/UdpMatcherTest.java

```diff
@@ -62,6 +65,28 @@ public void testReceiveResponseAcceptsResponseWithoutCorrelationInformation() {
 	}
 
 	@Test
+	public void testCancelledRequestShouldRemoveFromTokenMap() {
```

Comment:
We should improve this test case to better cover the type of request we are targeting, i.e. we need to make sure that we have received a separate ACK before canceling the request, e.g.
````
...
Exchange matched = matcher.receiveEmptyMessage(emptyMessageFor(exchange.getCurrentRequest()));
assertNotNull(matched);
exchange.getRequest().cancel();
...
````